metrics/stmt/metric.go

```diff
@@ -0,0 +1,67 @@
+package stmt
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+)
+
+// stmtMetric is a metric in the stmt domain
+type stmtMetric interface {
+	// Gets the name of the metric
+	Name() string
+	// Collects the metric
+	CollectMetric(ctx context.Context, db *sql.DB) (float64, error)
+}
+
+// Parent metric
+type metric struct {
+	name string
+}
+
+func (m metric) Name() string {
+	return m.name
+}
+
+type oldestQuery struct {
+	metric
+}
+
+func (q oldestQuery) CollectMetric(ctx context.Context, db *sql.DB) (float64, error) {
+	var t float64
+
+	query := `SELECT TIMER_WAIT FROM performance_schema.events_statements_current 
```

Comment:
https://hackmysql.com/golang/go-antipatterns/#non-variable-variable

This one and the other one, too.

Reply:
Go with proud: I've seen it many times before; yours was the one that put it on the board, so to speak.